### ISSUE TITLE
Wiki edit requests over 500 KiB rejected for being too large

### ISSUE THREAD

**Issue by TheEnigmaBlade** [reddit/reddit#1557]
Despite `/wiki/usernotes` being given [a size limit of 1 MB](https://github.com/reddit/reddit/blob/master/r2/r2/models/wiki.py#L84), submitting a request over 500 KiB/512 KB results in the following 413 error:

![](http://i.imgur.com/ikhoYNO.png)

I tested by directly editing `/wiki/usernotes` and found [any request over 512 KB is rejected](http://i.imgur.com/VLm2diV.png). No error is shown on the wiki interface, which I assume is due to the error being returned as HTML rather than JSON for errors [raised further down the line](https://github.com/reddit/reddit/blob/master/r2/r2/models/wiki.py#L379).

This error is returned by [the `LimitUploadSize` middleware](https://github.com/reddit/reddit/blob/9c680bdd1d6b5f95e423d509384a5f3c7864d515/r2/r2/config/middleware.py#L368), which I'm assuming it's intercepting and rejecting the request before reaching the wiki model.

I'm not submitting a pull request for now because I'm not sure which approach you would prefer to take and what negative consequences they could have in other parts of the site.

tl;dr: Usernotes are broken and I can't fix them this time. :heart:
